extensions/lifecycle/bucketProcessor/LifecycleBucketProcessor.js

```diff
@@ -328,6 +373,129 @@ class LifecycleBucketProcessor {
         });
     }
 
+    /**
+     * Connects to MongoDB using and retreives
+     * the location status collection
+     * @param {function} cb callback
+     * @returns {undefined} undefined
+     */
+    _setupMongoClient(cb) {
```

Comment:
it seems to me the whole mongodb change stream / status management could easily be extracted and made reusable : just need to eventually be able to call "resume" and "suspend" functions on the processor (possibly via callbacks)